.github/workflows/test-deps.yml

```diff
@@ -0,0 +1,89 @@
+        strategy:
+            fail-fast: false
+            matrix:
+              # "macos-latest" removed for now as Docker action is slow
+              os: [ "ubuntu-latest" ]  # TODO: add "windows-latest"
+              python-version: ["3.8", "3.9", "3.10"]
+        defaults:
+            run:
+              shell: bash
+        steps:
+            - name: Check out repository
+              uses: actions/checkout@v3
+
+            - name: Set up Python ${{ matrix.python-version }}
+              id: setup-python
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ matrix.python-version }}
+
+            - name: Install core dependencies only
+              run: python -m pip install .
+
+            # TODO: vendor only the parts that we need from the following action
+            - name: Docker
+              if: ${{ matrix.os == 'macos-latest' }}
+              uses: douglascamata/setup-docker-macos-action@5f13b5b88a09467c10f8462810902b60ff70fb75
+
+            # This checks that all modules can be imported correctly
+            - name: test imports
+              run: python -m pytest test/unittest/test_impall.py
+
+            - name: test CLI
+              run: python -m superduperdb
+
+    test-devdeps:
+        runs-on: ${{ matrix.os }}
+        strategy:
+            fail-fast: false
+            matrix:
+            # "macos-latest" removed for now as Docker action is slow
+              os: [ "ubuntu-latest" ]  # TODO: add "windows-latest"
+              python-version: ["3.8", "3.9", "3.10"]
+        defaults:
+            run:
+              shell: bash
+        steps:
+            - name: Check out repository
+              uses: actions/checkout@v3
+
+            - name: Set up Python ${{ matrix.python-version }}
+              id: setup-python
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ matrix.python-version }}
+
+            - name: Install developer dependencies
+              run: python -m pip install requirements/requirements-dev.in
+
+            # TODO: vendor only the parts that we need from the following action
+            - name: Docker
+              if: ${{ matrix.os == 'macos-latest' }}
+              uses: douglascamata/setup-docker-macos-action@5f13b5b88a09467c10f8462810902b60ff70fb75
+
+            - name: lint and test
+              run: |
+                make lint
+                make test
+
+    slack-notify:
+        needs: [test-nodevdeps, test-devdeps]
+        runs-on: ubuntu-latest
+        # https://stackoverflow.com/questions/71430668/how-to-run-a-github-actions-job-on-workflow-failure
+        if: ${{ always() && contains(needs.*.result, 'failure') }}
+        steps:
+            - name: Send GitHub Action trigger data to Slack workflow
+              uses: slackapi/slack-github-action@v1.24.0
+              env:
+                  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

Comment:
**(suggestion, non-blocking)** Would you be open to considering the slack [github bot integration](https://github.com/slackapi/slack-github-action)? That way, we won't need to maintain two integration APIs: we can use slack bot to notify us of new issues and PRs, as well as use it as an alert triggered by CI pipelines.

The advantage for CI: the notification destination (i.e. slack channel) is defined in the CI job instead of slack admin panel.